src/internal/colors/RgbwwColor.h

```diff
@@ -0,0 +1,307 @@
+/*-------------------------------------------------------------------------
+RgbwwColor provides a color object that can be directly consumed by NeoPixelBus
+
+Written by Michael C. Miller.
+
+I invest time and resources providing this open source code,
+please support me by dontating (see https://github.com/Makuna/NeoPixelBus)
+
+-------------------------------------------------------------------------
+This file is part of the Makuna/NeoPixelBus library.
+
+NeoPixelBus is free software: you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as
+published by the Free Software Foundation, either version 3 of
+the License, or (at your option) any later version.
+
+NeoPixelBus is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public
+License along with NeoPixel.  If not, see
+<http://www.gnu.org/licenses/>.
+-------------------------------------------------------------------------*/
+#pragma once
+
+struct RgbColor;
+struct HslColor;
+struct HsbColor;
+
+// ------------------------------------------------------------------------
+// RgbwwColor represents a color object that is represented by Red, Green, Blue
+// component values and an extra White component.  It contains helpful color 
+// routines to manipulate the color.
+// ------------------------------------------------------------------------
+struct RgbwwColor : RgbColorBase
+{
+    typedef NeoRgbwCurrentSettings SettingsObject;
+
+    // ------------------------------------------------------------------------
+    // Construct a RgbwwColor using R, G, B, W values (0-255)
+    // ------------------------------------------------------------------------
+    RgbwwColor(uint8_t r, uint8_t g, uint8_t b, uint8_t ww = 0, uint8_t cw = 0) :
+        R(r), G(g), B(b), WW(ww), CW(cw)
+    {
+    };
+
+    // ------------------------------------------------------------------------
+    // Construct a RgbColor using a single brightness value (0-255)
+    // This works well for creating gray tone colors
+    // (0) = black, (255) = white, (128) = gray
+    // ------------------------------------------------------------------------
+    RgbwwColor(uint8_t brightness) :
+        R(0), G(0), B(0), WW(brightness), CW(brightness)
```

Comment:
setting both to the same, is this double brightness?  Should this be 50% to each?  Is there a nonlinear equation that should be used?